sdk/core/azure-core/src/main/java/com/azure/core/util/BinaryData.java

```diff
@@ -593,6 +594,21 @@ public InputStream toStream() {
         return new ByteArrayInputStream(this.data);
     }
 
+    /**
+     * Returns a read-only {@link ByteBuffer} representation of this {@link BinaryData}.
+     * <p>
+     * Attempting to mutate the returned {@link ByteBuffer} will throw a {@link ReadOnlyBufferException}.
+     *
+     * <p><strong>Get a read-only ByteBuffer from the BinaryData</strong></p>
+     *
+     * {@codesnippet com.azure.util.BinaryData.toByteBuffer}
+     *
+     * @return A read-only {@link ByteBuffer} representing the {@link BinaryData}.
+     */
+    public ByteBuffer toByteBuffer() {
+        return ByteBuffer.wrap(this.data).asReadOnlyBuffer();
```

Comment:
Can this be done just once and cache the ByteBuffer instead of wrapping the data each time?

Reply:
It could be cached but each return would need to be a `duplicate` (another `ByteBuffer` view on top of the current one) and that would effectively be the same cost as this.